e2e/cypress/integration/notifications/desktop_notifications_3_spec.js

```diff
@@ -0,0 +1,109 @@
+
+    it('MM-T489_1 Display teammate nickname when nickname exists', () => {
+        cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+            spyNotificationAs('withNotification', 'granted');
+
+            // # Ensure notifications are set up to fire a desktop notification if are mentioned
+            changeDesktopNotificationAs('#desktopNotificationMentions');
+
+            // # Ensure display settings are set to "Show nickname if one exists, otherwise show first and last name"
+            changeTeammateNameDisplayAs('#name_formatFormatB');
+
+            const actualMsg = `@${testUser.username} first`;
+            const expected = `@${otherUser.nickname}: ${actualMsg}`;
+
+            // # Have another user send a post.
+            cy.postMessageAs({sender: otherUser, message: actualMsg, channelId: channel.id});
+
+            // * Desktop notification should be received with expected body.
+            cy.wait(TIMEOUTS.HALF_SEC);
+            cy.get('@withNotification').should('have.been.calledWithMatch', channel.display_name, (args) => {
+                expect(args.body, `Notification body: "${args.body}" should match: "${expected}"`).to.equal(expected);
+                return true;
+            });
+        });
+    });
+
+    it('MM-T489_2 Display teammates first and last name when nickname does not exists', () => {
+        cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+            // # Ensure other user is without nickname set up
+            patchUser(otherUser.id, {nickname: ''});
+
+            spyNotificationAs('withNotification', 'granted');
+
+            // # Ensure notifications are set up to fire a desktop notification if are mentioned
+            changeDesktopNotificationAs('#desktopNotificationMentions');
+
+            // # Ensure display settings are set to "Show nickname if one exists, otherwise show first and last name"
+            changeTeammateNameDisplayAs('#name_formatFormatB');
+
+            const actualMsg = `@${testUser.username} second`;
+            const expected = `@${otherUser.first_name} ${otherUser.last_name}: ${actualMsg}`;
+
+            // # Have another user send a post.
+            cy.postMessageAs({sender: otherUser, message: actualMsg, channelId: channel.id});
+
+            // * Desktop notification should be received with expected body.
+            cy.wait(TIMEOUTS.HALF_SEC);
+            cy.get('@withNotification').should('have.been.calledWithMatch', channel.display_name, (args) => {
+                expect(args.body, `Notification body: "${args.body}" should match: "${expected}"`).to.equal(expected);
+                return true;
+            });
+        });
+    });
+});
+
+function patchUser(userId, data) {
+    const sysadmin = getAdminAccount();
+
+    return cy.externalRequest({
+        user: sysadmin,
+        method: 'PUT',
+        path: `users/${userId}/patch`,
+        data,
+    }).its('status').should('eq', 200);
+}
```

Comment:
Same here. (Extracted this out to eliminate flakiness when dealing with stubbed notification)